### PR TITLE
eds.measurement improvements

### PR DESCRIPTION
**Pull request by svittoz**
## Description

- add **table processing** in eds.measurements
- add the possibility to extract **all measurements**
- update eds.measurements config to extract **more units**

## Checklist

- [ ] If this PR is a bug fix, the bug is documented in the test suite.
- [ ] Changes were documented in the changelog (pending section).
- [ ] If necessary, changes were made to the documentation (eg new pipeline).

### CODE REVIEW

File: edsnlp/pipes/misc/measurements/patterns.py [hunk 1]
```diff
@@ -624,5 +186,193 @@
...
 ]
+
+
+units_config = {
+    "g": {
+        "dim": "mass",
+        "degree": 1,
+        "scale": 1,
+        "terms": ["gramme", "grammes", "gr", "g"],
+        "per": ["g-1", "gr-1", "gr⁻¹", "g⁻¹"],
```
Comment (percevalw) [inline edsnlp/pipes/misc/measurements/patterns.py:201]:
```suggestion
        "per_terms": ["g-1", "gr-1", "gr⁻¹", "g⁻¹"],
```

**Review by percevalw — CHANGES_REQUESTED** [review]
Good work on the `use_tables` feature and the expansion of unit configuration!
However, I have mixed feelings about the way these new units are generated. On the one hand, there's a lot of redundancy (g -> kg, mg, cg, dc, etc), which suggests a factorisation of these prefixes. On the other hand, the devil is in the detail and there are many exceptions that require overriding these pre-generated units (cc for ‘cubic centimeter’, ‘d ui’ for deci ui) that have been forgotten in the course of these changes.

Replacing the unit terms after the large config doesn't help readability, so I'd be more in favour of wrapping the config individually and finding a way to place overrides near the base config, eg.

```python
    ...
    **generate_scaled_units(base_config, scale_prefixes, overrides)
```

```python
config.update(generate_scaled_units(base_config, scale_prefix))
config["cm3"]...
```

File: edsnlp/pipes/misc/measurements/patterns.py [hunk 2]
```diff
@@ -1,3 +1,75 @@
...
+        scale_value = 1 / scale_value
+        scaled_key = f"per_{scale_prefix[0]}{key[4:]}"
+    else:
+        scaled_key = f"{scale_prefix[0]}{key}"
+
+    return {
+        scaled_key: {
+            "dim": config["dim"],
+            "degree": config["degree"],
+            "scale": scale_value,
```
Comment (percevalw) [inline edsnlp/pipes/misc/measurements/patterns.py:38]: In the case of "liters", I believe this doesn't account for the fact that a liter is 1e-3 m3.
And while we have m3 -> 1, dm3 -> 1e-3, cm3 -> 1e-6, we have liter -> 1e-3, deciliter -> 1e-4, centiliter -> 1e-5.
Managing the scales is not easily done automatically...

**Comment by svittoz** [comment]
> Good work on the `use_tables` feature and the expansion of unit configuration! However, I have mixed feelings about the way these new units are generated. On the one hand, there's a lot of redundancy (g -> kg, mg, cg, dc, etc), which suggests a factorisation of these prefixes. On the other hand, the devil is in the detail and there are many exceptions that require overriding these pre-generated units (cc for ‘cubic centimeter’, ‘d ui’ for deci ui) that have been forgotten in the course of these changes.
> 
> Replacing the unit terms after the large config doesn't help readability, so I'd be more in favour of wrapping the config individually and finding a way to place overrides near the base config, eg.
> 
> ```python
>     ...
>     **generate_scaled_units(base_config, scale_prefixes, overrides)
> ```
> 
> ```python
> config.update(generate_scaled_units(base_config, scale_prefix))
> config["cm3"]...
> ```

**Pull request merged**